### ISSUE TITLE
Pandas enforces subplot_adjust creating UserWarning with constrained_layout

### ISSUE THREAD

**Issue by michaelaye** [matplotlib/matplotlib#16464]
### Bug report

**Bug summary**

Pandas dataframe plotting uses subplot_adjust, always creating a UserWarning because I am using constrained_layout as my default layouter.

Cross-issue with https://github.com/pandas-dev/pandas/issues/25261 (I'm uncertain where this should/could be fixed).

**Code for reproduction**

```python
import matplotlib as mpl
mpl.rcParams['figure.constrained_layout.use'] = True
tseries = pd.date_range('now', freq='s', periods=100)
df = pd.DataFrame({'data': np.random.random(100)}, index=tseries)
df.plot()
```

**Actual outcome**

<img width="969" alt="Screenshot 2020-02-10 09 55 00" src="https://user-images.githubusercontent.com/69774/74177110-4df28980-4bf6-11ea-89be-8b02086ef03b.png">

**Expected outcome**

No UserWarning

**Matplotlib version**

  * Operating system: macOS 10.14.6
  * Matplotlib version: 3.1.3
  * Matplotlib backend (`print(matplotlib.get_backend())`): 'module://ipykernel.pylab.backend_inline'
  * Python version: 3.7.6
  * Jupyter version (if applicable): Jupyterlab 1.2.6
  * Other libraries: 

All installed via conda-forge.

**Comment by jklymak** [comment]
This is how `constrained_layout` is supposed to work - if the user manually adjusts the subplots, it stops trying to do things automatically.

**Comment by michaelaye** [comment]
With user you mean pandas here, I guess?

**Comment by jklymak** [comment]
Any user or any package calling `subplots_adjust`.

**Comment by michaelaye** [comment]
Ok. My point here is that it's out of the user's hand. Pandas just does the subplots_adjust, therefore any user of the constrained_layout feature constantly gets these UserWarnings.

**Comment by jklymak** [comment]
Matplotlib is working as designed, so I guess the question is what change would you hope for on our end?  The only two I can think of are a) we don't emit a `UserWarning`, but I think the warning is needed in this case, or b) we ignore `subplots_adjust`.  But I think we can't really do that as people have perfectly valid reasons for manually adjusting their plots, and don't want some more general setting overriding the more specific.  

From the panda's side they could not call `subplots_adjust` if `constrained_layout` is True.

**Comment by michaelaye** [comment]
I tend to agree, but the matplotlib twitter account encouraged me to create this issue and link it to the pandas' one. ;)

**Comment by story645** [comment]
I figured this is a common enough bug that it's useful to be documented here 🤷‍♀
